libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1729,31 +1730,80 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
 		requireLValue(_operation.subExpression(), false);
 	else
 		_operation.subExpression().accept(*this);
-	Type const* subExprType = type(_operation.subExpression());
-	TypeResult result = type(_operation.subExpression())->unaryOperatorResult(op);
-	if (!result)
+	Type const* operandType = type(_operation.subExpression());
+
+	// Check if the operator is built-in or user-defined.
+	TypeResult builtinResult = operandType->unaryOperatorResult(op);
+	set<FunctionDefinition const*> matchingDefinitions = operandType->operatorDefinitions(
```

Comment:
i.e. easiest quickest fix for https://github.com/ethereum/solidity/pull/13790/files#r1091037315 would probably be using AST-ID ordering for this set.

Reply:
Same here. Changed to `vector`.